src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private class CpuUtilizationReader
+        {
+            private struct ProcessCpuInformation
+            {
+                public ulong idleTime;
+                public ulong kernelTime;
+                public ulong userTime;
+            }
+
+            private ProcessCpuInformation _cpuInfo = new ProcessCpuInformation();
+
+            public CpuUtilizationReader()
+            {
+                GetCpuUtilization(); // Call once to initialize the usage buffer
+            }
+
+            private unsafe int GetCpuUtilization()
+            {
+                if (!Interop.Kernel32.GetSystemTimes(out var idleTime, out var kernelTime, out var userTime))
+                {
+                    int error = Marshal.GetLastWin32Error();
+                    var exception = new OutOfMemoryException();
+                    exception.SetErrorCode(error);
+                    throw exception;
+                }
+
+                ulong cpuTotalTime = (userTime - _cpuInfo.userTime) + (kernelTime - _cpuInfo.kernelTime);
+                ulong cpuBusyTime = cpuTotalTime - (idleTime - _cpuInfo.idleTime);
+
+                _cpuInfo.kernelTime = kernelTime;
+                _cpuInfo.userTime = userTime;
+                _cpuInfo.idleTime = idleTime;
+
+                if (cpuTotalTime > 0)
```

Comment:
This check would have caught negative values before when it was signed. Could make cpuTotalTime and cpuBusyTime signed. Should also check that `cpuBusyTime >= 0`.